### PR TITLE
wormhole-william: init at v1.0.4

### PR DESCRIPTION
**Pull request by psanford**
###### Motivation for this change

Add package for `wormhole-william` end-to-end encrypted file transfer tool.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/wormhole-william/default.nix
Comment (danieldk) [inline pkgs/tools/networking/wormhole-william/default.nix:20]: `meta.description` should not have a period at the end, could you remote it?

**Review by danieldk — CHANGES_REQUESTED** [review]
Thank you for your first contribution to nixpkgs! This PR already looks very good.

One small request. Also, could you split up the commit in two, one that adds the package and one (ordered before it) that adds you to `maintainers.nix` with the following message:

```
maintainers: add psanford
```

File: pkgs/tools/networking/wormhole-william/default.nix
Comment (psanford) [reply pkgs/tools/networking/wormhole-william/default.nix]: Fixed.

**Comment by psanford** [comment]
Thanks for reviewing @danieldk! I've updated the PR to address your comments.

**Review by danieldk — APPROVED** [review]
Thanks a lot for making the changes, looks good now!

Result of `nixpkgs-review pr 93030` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- wormhole-william
</details>

Will merge it if ofborg is also happy.

File: pkgs/tools/networking/wormhole-william/default.nix
Comment (zowoq) [inline pkgs/tools/networking/wormhole-william/default.nix:7]: Not needed with `buildGoModule`.

```suggestion
```

Comment (psanford) [reply pkgs/tools/networking/wormhole-william/default.nix]: Thanks, I've removed this!

**Pull request merged**
